Unused variable in Z3Interface.cpp causes error-as-warning error when compiling with MSVC

On @axic 's request.

Solidity: `Version: 0.4.18-develop.2017.10.18+commit.fda8499c.Windows.msvc` (after fixing)

```
C:\Users\andreas\solidity\libsolidity\formal\Z3Interface.cpp(101): error C2220: warning treated as error - no 'object' fil e generated [C:\Users\andreas\solidity\build\libsolidity\solidity.vcxproj]
C:\Users\andreas\solidity\libsolidity\formal\Z3Interface.cpp(101): warning C4101: '_e': unreferenced local variable [C:\Us ers\andreas\solidity\build\libsolidity\solidity.vcxproj]
```

The cause is an unused error in a catch clause. This only happens because I manually linked with Z3, so it should not affect other users or CI, unless it is if left as-is when Z3 is included in windows builds by default (maybe it already is).

My suggestion was to just disable the warning in the cmake script (`/wd4101`).

--- a/libsolidity/formal/Z3Interface.cpp
+++ b/libsolidity/formal/Z3Interface.cpp
@@ -98,7 +98,7 @@ pair<CheckResult, vector<string>> Z3Interface::check(vector<Expression> const& _
 				values.push_back(toString(m.eval(toZ3Expr(e))));
 		}
 	}
-	catch (z3::exception const& _e)
+	catch (z3::exception const&)
 	{
 		result = CheckResult::ERROR;
 		values.clear();

